### PR TITLE
makeself: common build, patch instead of sed

### PR DESCRIPTION
**Pull request by innerspacepilot**

### CODE REVIEW

File: srcpkgs/makeself/template [hunk 1]
```diff
@@ -1,34 +1,23 @@
...
-	# rename makeself-header.sh to makeself-header inside makeself.sh because it will be renamed later
-	sed -i 's/^HEADER=\`dirname "$0"\`\/makeself-header.sh$/HEADER=\`dirname "$0"\`\/makeself-header/' makeself.sh
-
 	vmkdir usr/share/man/man1
 	for f in makeself makeself-header; do
 		vbin ${f}.sh ${f}
 	done
 	install -m644 *.1 ${DESTDIR}/usr/share/man/man1
 }
+
```
Comment (congdanhqx) [inline srcpkgs/makeself/template]: Please this empty line.

Comment (innerspacepilot) [reply srcpkgs/makeself/template]: fixed

File: srcpkgs/makeself/template [hunk 2]
```diff
@@ -1,31 +1,19 @@
 # Template file for 'makeself'
 pkgname=makeself
 version=2.4.0
-revision=2
+revision=3
 archs=noarch
-create_wrksrc=yes
+wrksrc="${pkgname}-release-${version}"
 makedepends="tar gzip bzip2"
```
Comment (congdanhqx) [inline srcpkgs/makeself/template]: Is this necessary?
Most (all?) of our packages require `tar(1)` and either `gzip(1)` or `bzip2(1)` or `xz(1)` to extract.
And we don't need to specify them as `makedepends`.
`xbps-install(1)` also requires them to run but `srcpkgs/xbps/template` doesn't have them.

Anyway, if those packages should be included here, how about `xz(1)` and other compression package? And package for cryptographic options?

---
Sorry, but I would like to have conscious from the team.

Comment (innerspacepilot) [reply srcpkgs/makeself/template]: I don't know actually, I've made minimal changes and left other stuff as it was before.

Comment (Chocimier) [reply srcpkgs/makeself/template]: @congdanhqx: Good point! Just one detail missed: tar is always installed as `hostmakedepends`.
@innerspacepilot: Thanks for contribution!
As you are modifying template, remove makedepends as they are not needed and depends as these packages are required by base system and please replace `$pkgname` with `makeself` in all template.
If anything non-standard (like xz) is needed at runtime, you can add it to depends.

Comment (innerspacepilot) [reply srcpkgs/makeself/template]: done

**Pull request merged**